### PR TITLE
nasm: bump to 2.16.01

### PR DESCRIPTION
**Pull request by hovercats**
bump nasm to 2.16.01

### CODE REVIEW

File: pkg/nasm/config.h [hunk 1]
```diff
@@ -160,8 +163,8 @@
 #define HAVE___BUILTIN_BSWAP16 1
 #define HAVE___BUILTIN_BSWAP32 1
 #define HAVE___BUILTIN_BSWAP64 1
-/* probe HAVE___BUILTIN_CHOOSE_EXPR */
-/* probe HAVE___BUILTIN_CLZ */
+#define HAVE___BUILTIN_CHOOSE_EXPR 1
+#define HAVE___BUILTIN_CLZ 1
```
Comment (rnpnr) [inline pkg/nasm/config.h:167]: I think these should remain commented out. They are checked separately by probe: 
https://github.com/oasislinux/oasis/blob/master/probe/gen.lua#L13

**Comment by hovercats** [comment]
Youre absolutely right. Fixing this right away.

File: pkg/nasm/config.h (hunk 1 above)
Comment (michaelforney) [reply pkg/nasm/config.h]: Thanks for catching this, @rnpnr!

**Comment by michaelforney** [comment]
Applied, thanks!

**Pull request closed without merging**